File: problems_1_50/multiples_3_5/answer.ts
function findModValues(a: number, b: number, size: number): number[]{
  const answers: number[] = [];
  for(let i = 0; i < size; ++i) {
    if( i % a === 0 || i % b === 0){
      answers.push(i);
    }
  }
  return answers;
}

const a: number = 3;
const b: number = 5
const size: number = 1_000;
const values: number[] = findModValues(a, b, size);
const answer: number = values
  .reduce( (a: number, b: number) => a + b, 0 );
console.log(`Sum of multiples of ${a} and ${b} less than ${size} is ${answer}`);

File: problems_1_50/multiples_3_5/answer.rs
fn populate_mod_vector(a: i32, b: i32, size: i32) -> Vec<i32> {
    let mut vec = vec![];
    for i in 1..size {
        if i % a == 0 || i % b == 0 {
            vec.push(i)
        }
    }
    vec
}

fn main() {
    let vec = populate_mod_vector(3, 5, 1_000);
    let mut sum = 0;

    for x in vec.iter() {
        sum += x;
    }

    println!(
        "Sum of multiples of {} and {} less than {} is {}",
        3,
        5,
        1_000,
        sum
    )
}
